control_plane/Cargo.toml

```diff
@@ -24,9 +26,10 @@ tar.workspace = true
 thiserror.workspace = true
 toml.workspace = true
 tokio.workspace = true
+tokio-postgres.workspace = true
+tokio-util.workspace = true
 url.workspace = true
-# Note: Do not directly depend on pageserver or safekeeper; use pageserver_api or safekeeper_api
-# instead, so that recompile times are better.
+pageserver = { path = "../pageserver" }
```

Comment:
Just checked binary sizes with this change: `neon_local` and `attachment_service` go from being ~90MB to being ~220MB.  The literal binary size isn't important but it indicates the amoutn of extra linking work that's going to happen + generally slow down workflows.  Can we get all the API structs we need into `pageserver_api` instead?

Reply:
All for it, was expecting resistance to that.

Trying it out